### PR TITLE
Improve processing of declared license URLs

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: utils/src/main/kotlin/DeclaredLicenseProcessor.kt [hunk 1]
```diff
@@ -29,7 +29,8 @@ import com.here.ort.spdx.SpdxExpression
 object DeclaredLicenseProcessor {
     private val prefixesToRemove = listOf(
         "https://choosealicense.com/licenses/",
-        "https://opensource.org/licenses/"
+        "https://opensource.org/licenses/",
+        "https://www.gnu.org/licenses/old-licenses/"
```
Comment (tsteenbe) [inline utils/src/main/kotlin/DeclaredLicenseProcessor.kt]: @sschuberth Please also add https://licenses.nuget.org, https://tldrlegal.com/license and https://spdx.org/licenses

Comment (tsteenbe) [inline utils/src/main/kotlin/DeclaredLicenseProcessor.kt]: On 2nd thought I propose we remove "https://www.gnu.org/licenses/old-licenses/" here or change it to just "https://www.gnu.org/licenses"  else one looses in SPDXLicenseMapping the 'old-licenses' bit tells you that it's (L)GPL version 2 family and not 3.

File: utils/src/main/kotlin/DeclaredLicenseProcessor.kt [hunk 2]
```diff
@@ -29,15 +29,32 @@ import com.here.ort.spdx.SpdxExpression
...
+    )
+
+    private val suffixesToRemove = listOf(
+        ".dbk", ".html", ".md", ".odt", ".rtf", ".tex", ".txt"
```
Comment (tsteenbe) [inline utils/src/main/kotlin/DeclaredLicenseProcessor.kt]: Can you .php for http://www.opensource.org/licenses/mit-license.php?

File: utils/src/main/kotlin/DeclaredLicenseProcessor.kt (hunk 1 above)
Comment (sschuberth) [reply utils/src/main/kotlin/DeclaredLicenseProcessor.kt]: Done.

Comment (tsteenbe) [reply utils/src/main/kotlin/DeclaredLicenseProcessor.kt]: Add http://www.opensource.org/licenses/ as well

**Comment by sschuberth** [comment]
> Add http://www.opensource.org/licenses/ as well

@tsteenbe, adding http / www variants is actually not needed because of https://github.com/heremaps/oss-review-toolkit/pull/1807/commits/07cb0d0559c2a6c3fdfa2f31d24031a446910a5a.

**Pull request merged**